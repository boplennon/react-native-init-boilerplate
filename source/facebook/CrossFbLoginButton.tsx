import React from 'react';
import { LoginButton, AccessToken } from 'react-native-fbsdk';
import styles from '../styles';

/**
 * Uses the nativly linked Facebook SDK to log in
 *
 * https://github.com/facebook/react-native-fbsdk
 */
export const CrossFbLoginButton = () => (
    <LoginButton
    style={styles.button}
      onLoginFinished={(error, result) => {
        if (error) {
          console.log('Facebook login has error: ' + result.error);
        } else if (result.isCancelled) {
          console.log('Facebook login is cancelled.');
        } else {
          AccessToken.getCurrentAccessToken().then((data) => {
            if (data) console.log(data.accessToken.toString());
          });
        }
      }}
      onLogoutFinished={() => console.log('logout.')}
    />
);

export default CrossFbLoginButton;
