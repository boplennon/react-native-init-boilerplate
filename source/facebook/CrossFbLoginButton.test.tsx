/// <reference types="jest"/>
import React from 'react';
import TestRenderer from 'react-test-renderer';
import CrossFbLoginButton from './CrossFbLoginButton';
import { LoginButton } from 'react-native-fbsdk';
import styles from '../styles';


function setup() {
  return TestRenderer.create(<CrossFbLoginButton />);
}

describe('components', () => {
  describe('<CrossFbLoginButton />', () => {
    const  wrapper = setup();

    it('should match snapshot', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should have button style', () => {
      const button = wrapper.root.findByType(LoginButton);
      expect(button.props.style).toEqual(styles.button);
    });
  });
});
