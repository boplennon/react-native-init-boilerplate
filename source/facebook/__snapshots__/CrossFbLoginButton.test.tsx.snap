// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <CrossFbLoginButton /> should match snapshot 1`] = `
<Button
  onLoginFinished={[Function]}
  onLogoutFinished={[Function]}
  style={
    Object {
      "height": 45,
      "marginBottom": 10,
      "marginRight": 10,
      "marginTop": 10,
    }
  }
/>
`;
